### PR TITLE
Add `@override` for subclasses of `Accelerator`

### PR DESCRIPTION
**Pull request by VictorPrins**
## What does this PR do?

It adds the `@override` decorator for all subclasses of `lightning.pytorch.accelerators.Accelerator` (https://github.com/Lightning-AI/lightning/issues/18695).

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
:books: Documentation preview :books:: https://pytorch-lightning--18934.org.readthedocs.build/en/18934/

### CODE REVIEW

**Comment by VictorPrins** [comment]
cc @carmocca @awaelchli This is ready for merging.

File: src/lightning/pytorch/accelerators/cuda.py [hunk 1]
```diff
@@ -73,29 +77,35 @@ def get_device_stats(self, device: _DEVICE) -> Dict[str, Any]:
...
         """Get the devices when set to auto."""
         return num_cuda_devices()
 
     @staticmethod
+    @override
     def is_available() -> bool:
         return num_cuda_devices() > 0
 
     @classmethod
+    @override
```
Comment (carmocca) [inline src/lightning/pytorch/accelerators/cuda.py:108]: Does the order of staticmethod/classmethod and override matter? In https://github.com/Lightning-AI/lightning/pull/18911/files#diff-0771b30c8f08ba845b153fb4c07cb219bb5648bed2abebdb95afd6def13b830eR881-R882 you used a different order

Comment (VictorPrins) [reply src/lightning/pytorch/accelerators/cuda.py:108]: Good catch; PEP-698 states 

> `@override` needs to execute before many special descriptor-based decorators like `@property`, `@staticmethod`, and `@classmethod`.

This means that `@override` should go _below_ those decorators.

However:

> The `@override` decorator needs to execute after ordinary decorators like `@functools.lru_cache` that use wrapper functions.

So `@override` should go _above_ these decorators.

In any case, it doesn't matter for any of the CI tests and PEP-698 says "we do not expect type checkers to validate the ordering of decorators".

Nevertheless, I will be consistent and modify the other PR. The ordering in this PR is the correct one.

**Review by carmocca — APPROVED** [review]
Very good!

**Comment by VictorPrins** [comment]
@awaelchli ready to merge

**Review by awaelchli — APPROVED** [review]

**Pull request merged**